L1TriggerConfig/L1TConfigProducers/src/L1TMuonOverlapFwVersionOnlineProd.cc

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <fstream>
+
+#include "CondTools/L1TriggerExt/interface/L1ConfigOnlineProdBaseExt.h"
+#include "CondFormats/L1TObjects/interface/L1TMuonOverlapFwVersion.h"
+#include "CondFormats/DataRecord/interface/L1TMuonOverlapFwVersionRcd.h"
+#include "CondFormats/DataRecord/interface/L1TMuonOverlapFwVersionO2ORcd.h"
+#include "L1Trigger/L1TCommon/interface/TriggerSystem.h"
+#include "L1Trigger/L1TCommon/interface/XmlConfigParser.h"
+#include "OnlineDBqueryHelper.h"
+
+#include "xercesc/util/PlatformUtils.hpp"
+using namespace XERCES_CPP_NAMESPACE;
+
+class L1TMuonOverlapFwVersionOnlineProd : public L1ConfigOnlineProdBaseExt<L1TMuonOverlapFwVersionO2ORcd, L1TMuonOverlapFwVersion> {
+private:
+  const bool transactionSafe;
+  const edm::ESGetToken<L1TMuonOverlapFwVersion, L1TMuonOverlapFwVersionRcd> baseSettings_token;
+
+public:
+  std::unique_ptr<const L1TMuonOverlapFwVersion> newObject(const std::string& objectKey, const L1TMuonOverlapFwVersionO2ORcd& record) override;
+
+  L1TMuonOverlapFwVersionOnlineProd(const edm::ParameterSet&);
+  ~L1TMuonOverlapFwVersionOnlineProd(void) override {}
+};
+
+void replaceAll(std::string& str, const std::string& from, const std::string& to) {
+  if(from.empty())
+    return;
+  size_t start_pos = 0;
+  while((start_pos = str.find(from, start_pos)) != std::string::npos) {
+    str.replace(start_pos, from.length(), to);
+    start_pos += to.length();
+  }
+}
+
+void removeAll(std::string& str, const std::string& from, const std::string& to) {
+  if(from.empty())
+    return;
+  size_t start_pos = 0;
+  while((start_pos = str.find(from, start_pos)) != std::string::npos) {
+    size_t end_pos = str.find(to) + to.length();
+    int length = end_pos - start_pos;
+    str.replace(start_pos, length, "");
+  }
+}
+
+L1TMuonOverlapFwVersionOnlineProd::L1TMuonOverlapFwVersionOnlineProd(const edm::ParameterSet& iConfig)
+    : L1ConfigOnlineProdBaseExt<L1TMuonOverlapFwVersionO2ORcd, L1TMuonOverlapFwVersion>(iConfig),
+    transactionSafe(iConfig.getParameter<bool>("transactionSafe")),
+    baseSettings_token(wrappedSetWhatProduced(iConfig).consumes()) {}
+
+std::unique_ptr<const L1TMuonOverlapFwVersion> L1TMuonOverlapFwVersionOnlineProd::newObject(
+    const std::string& objectKey, const L1TMuonOverlapFwVersionO2ORcd& record) {
+
+  const L1TMuonOverlapFwVersionRcd& baseRcd = record.template getRecord<L1TMuonOverlapFwVersionRcd>();
+  auto const& baseSettings = baseRcd.get(baseSettings_token);
```

Comment:
ES get() is deprecated, move to the new method with the tokens and `getData` / `getHandle`. There is 4 occurance of `.get(`, so you need to resolve all of those

Reply:
Not sure what is wrong here.  I am using a token to get data from the record, like suggested in
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_a_Record
Is this not allowed anymore?

Reply:
Ah yes, you are right, you are getting this from the record and not the ES. In this case this is fine, sorry for the noise!

Reply:
Feel free to mark this comment "resolved"

Reply:
On the other hand there are cases when you do take it from the ES and thus it's wrong, let me point to another example

Reply:
Here https://github.com/cms-sw/cmssw/pull/35876/files/250d619bf5e78fa43e2b76a77b872ab6993d3d64#r747562872